recursion_and_iteration.rb

```diff
@@ -0,0 +1,59 @@
+def factorial_r(num)
+  if num == 0
+    1
+  else
+    num * factorial_r(num - 1)
+  end
+end
+
+def factorial_i(num)
+  big_number = num
+  if num == 0
+    big_number = 1
+  else
+    while num > 1    
+      big_number = big_number * num
+      num -= 1
+    end
+  end
+  puts big_number
+end
+
+def sumdig_r(num)
+  if num < 10
+    num
+  else
+    num % 10 + schwadulo(num/10)
+  end
+end
+
+def sumdig_i(num)
+  iterator = 10
+  answer = 0
+  until num == 0
+    answer += num % 10
+    num = num / 10
+  end
+  puts answer
+end
+
+def palindrome_r(string)
+  if string.length == 0 || string.length == 1
+    true
+  else
+    if string[0] == string[-1] && palindrome_r(string[1..-2])
+      return true
+    end
+  end
+end

```

Comment:
It looks like this only returns `true` or `nil` but never false. Instead of using `if` on line 44, just return the outcome of that conditional. You obviously figured out the recursive trick here!